### PR TITLE
Environment Variables for upgrade scenarios are now robottelo properties - Phase 2

### PR DESCRIPTION
**Pull request by jyejare**
Changes in Files wrt $subject:

```
- tests/upgrades/test_client.py
- tests/upgrades/test_remoteexecution.py
- tests/upgrades/test_repository.py
- tests/upgrades/test_subscription.py
- tests/upgrades/test_syncplan.py
- tests/upgrades/test_yum_plugins.py
```

Also, Some nitpicks and small improvements wrt manifest upload.

### CODE REVIEW

File: tests/upgrades/test_remoteexecution.py [hunk 1]
```diff
@@ -45,18 +43,18 @@ class Scenario_remoteexecution_external_capsule(APITestCase):
...
-        cls.vm_domain_name = os.environ.get('VM_DOMAIN')
+        cls.bridge = settings.vlan_networking.bridge
+        cls.subnet = settings.vlan_networking.subnet
+        cls.gateway = settings.vlan_networking.gateway
+        cls.netmask = settings.vlan_networking.netmask
+        cls.vm_domain_name = settings.upgrade.vm_domain
         cls.vm_domain = entities.Domain().search(query={'search': 'name="{}"'
                                                  .format(cls.vm_domain_name)})
-        cls.proxy_name = os.environ.get('RHEV_CAP_HOST')
+        cls.proxy_name = settings.upgrade.rhev_cap_host or settings.upgrade.capsule_hostname
```
Comment (omkarkhatavkar) [inline tests/upgrades/test_remoteexecution.py]: `capsule_hostname` exist in robottelo.properties file but need to be sed while running the pipeline, this currently not exists in  https://github.com/SatelliteQE/robottelo-ci/blob/c89f0f2026e94e02672086b90dbebb1b983157a4/scripts/satellite6-upgrade-run-scenarios.sh#L19

File: tests/upgrades/test_yum_plugins.py [hunk 2]
```diff
@@ -89,7 +89,6 @@ def _check_yum_plugins_count(self, client_container_id):
     def _check_package_installed(self, client_container_id, package):
         """Verify if package is installed on docker content host."""
         kwargs = {'host': self.docker_vm}
-        docker_execute_command
```
Comment (omkarkhatavkar) [inline tests/upgrades/test_yum_plugins.py]: why `docker_execute_command` was removed from the argument?

File: tests/upgrades/test_yum_plugins.py [hunk 3]
```diff
@@ -112,10 +111,8 @@ def _install_or_update_package(self, client_container_id, package, update=False)
 
     def _create_custom_rhel_tools_repos(self, product):
         """Install packge on docker content host."""
-        rhel_repo_url = os.environ.get('{}_CUSTOM_REPO'.format(self.client_os.upper()))
-
-        tools_repo_url = os.environ.get(
-            'TOOLS_{}'.format(self.client_os.upper()))
+        rhel_repo_url = settings.rhel7_os
+        tools_repo_url = settings.sattools_repo[DISTRO_RHEL7]
```
Comment (omkarkhatavkar) [inline tests/upgrades/test_yum_plugins.py]: here also we hardcoding to rhel7, earlier it was dynamically generated anything (rhel6,7,8). we need to have some dynamic nature or adding if else would make sense.

File: tests/upgrades/test_client.py [hunk 4]
```diff
@@ -48,10 +47,10 @@ def create_activation_key_for_client_registration(
     :return nailgun.entity.ActivationKey: Activation key
     """
     client_os = client_os.upper()
-    from_ver = os.environ.get('FROM_VERSION')
+    from_ver = settings.upgrade.from_version
     rhel_prod_name = 'scenarios_rhel{}_prod'.format(client_os[-1])
     rhel_repo_name = '{}_repo'.format(rhel_prod_name)
-    rhel_url = os.environ.get('{}_CUSTOM_REPO'.format(client_os))
+    rhel_url = settings.rhel7_os
```
[the remark this answers is not in the document]
Comment (jyejare) [reply tests/upgrades/test_client.py]: Again, I will give you the same answer given in my last phase-1 PR.

Actually, we setting an url for thel7 but all the validations are happening for rhel7 only and those are hardcoded.
When we will switch to RHEL8 as default all those tests will fail.

[the remark this answers is not in the document]
Comment (jyejare) [reply tests/upgrades/test_client.py]: I know we should do it for rhel8 as well. But then we should run the validations and setup twice, one for rhel7 and another for rhel8.

But that change is not part of this PR. We can handle that separately if we really want to test on rhel7 and rhel8.

File: tests/upgrades/test_remoteexecution.py (hunk 1 above)
Comment (jyejare) [reply tests/upgrades/test_remoteexecution.py]: This is a completely optional field and you can only use that field while running the tests locally on your local capsule.

In short, its either RHEV_CAP_HOST or CAPSULE_HOSTNAME

File: tests/upgrades/test_yum_plugins.py (hunk 2 above)
Comment (jyejare) [reply tests/upgrades/test_yum_plugins.py]: This was hanging without being used, I don't see that func call is doing anything. 

Also, I am not sure why r u calling it as an argument. I believe below this line, this func is being correctly called an argument.

File: tests/upgrades/test_yum_plugins.py (hunk 3 above)
Comment (jyejare) [reply tests/upgrades/test_yum_plugins.py]: Same as above.

File: tests/upgrades/test_remoteexecution.py (hunk 1 above)
Comment (omkarkhatavkar) [reply tests/upgrades/test_remoteexecution.py]: what if RHEV_CAP_HOST does not exist? you want to use CAPSULE_HOSTNAME if that is never a case then we can make it like this 

`cls.proxy_name = settings.upgrade.rhev_cap_host`

File: tests/upgrades/test_yum_plugins.py (hunk 2 above)
Comment (omkarkhatavkar) [reply tests/upgrades/test_yum_plugins.py]: my bad I did not notice that.

File: tests/upgrades/test_remoteexecution.py (hunk 1 above)
Comment (jyejare) [reply tests/upgrades/test_remoteexecution.py]: RHEV_CAP_HOST is there always when we run tests on Jenkins as we run upgrades on RHEV based images.

File: tests/upgrades/test_client.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (omkarkhatavkar) [reply tests/upgrades/test_client.py]: We can create an issue for RHEL8 and handle in separate.

File: tests/upgrades/test_remoteexecution.py (hunk 1 above)
Comment (omkarkhatavkar) [reply tests/upgrades/test_remoteexecution.py]: this means we don't require `CAPSULE_HOSTNAME` as an additional/optional argument, this where I'm pointing. If that is not true do we set  `CAPSULE_HOSTNAME` in the pipeline anywhere ?.

Comment (omkarkhatavkar) [reply tests/upgrades/test_remoteexecution.py]: you can only use that field while running the tests locally on your local capsule. - sorry I missed this message. In this case, looks ok then.

**Comment by jyejare** [comment]
Test Results:
------------------------

# test_repository.py
```
# py.test -s tests/upgrades/test_repository.py::Scenario_custom_repo_check -m 'pre_upgrade'
========================================== test session starts ==========================================
platform linux -- Python 3.6.5, pytest-4.0.2, py-1.8.0, pluggy-0.12.0
rootdir: /home/jenkins/workspace/automation-preupgrade-6.6-scenario-tests-rhel7, inifile:
plugins: xdist-1.27.0, forked-1.0.2, mock-1.10.4, services-1.3.1, cov-2.7.1
collected 2 items / 1 deselected
========================= 1 passed, 1 deselected, 6 warnings in 218.49 seconds ==========================
```

**Comment by vijay8451** [comment]
LGTM .. ack

**Pull request merged**